src/Symfony/Component/HttpClient/CachingHttpClient.php

```diff
@@ -106,41 +390,313 @@ public function request(string $method, string $url, array $options = []): Respo
+     * the Expires header for a valid timestamp, and if present, returns the
+     * difference between the timestamp and the current time minus the current age.
+     *
+     * If none of the above directives or headers are present, the method returns
+     * null.
+     *
+     * @param array<string, string|string[]> $headers      an array of HTTP headers
+     * @param array<string, string|true>     $cacheControl an array of parsed Cache-Control directives
+     *
+     * @return int|null the maximum age of the response, or null if it cannot be
+     *                  determined
+     */
+    private function determineMaxAge(array $headers, array $cacheControl): ?int
+    {
+        $age = $this->getCurrentAge($headers);
+
+        if ($this->sharedCache && isset($cacheControl['s-maxage'])) {
+            $val = (int) $cacheControl['s-maxage'];
+
+            return max(0, $val - $age);
+        }
+
+        if (isset($cacheControl['max-age'])) {
+            $val = (int) $cacheControl['max-age'];
+
+            return max(0, $val - $age);
+        }
+
+        foreach ($headers['expires'] ?? [] as $expire) {
+            $ts = strtotime($expire);
+            if (false !== $ts) {
+                $diff = $ts - time() - $age;
+
+                return max($diff, 0);
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Retrieves the current age of the response from the headers.
+     *
+     * @param array<string, string|string[]> $headers an array of HTTP headers
+     *
+     * @return int The age of the response in seconds. Defaults to 0 if not present.
+     */
+    private function getCurrentAge(array $headers): int
+    {
+        return (int) ($headers['age'][0] ?? 0);
+    }
+
+    /**
+     * Calculates the expiration time of the response given the maximum age.
+     *
+     * @param int|null $maxAge the maximum age of the response in seconds, or null if it cannot be determined
+     *
+     * @return int|null the expiration time of the response as a Unix timestamp, or null if the maximum age is null
+     */
+    private function calculateExpiresAt(?int $maxAge): ?int
+    {
+        if (null === $maxAge) {
+            return null;
+        }
+
+        return time() + $maxAge;
     }
 
-    public function reset(): void
+    /**
+     * Checks if the server response is cacheable according to the HTTP 1.1
+     * specification (RFC 9111).
+     *
+     * This function will return true if the server response can be cached,
+     * false otherwise.
+     *
+     * @param int                            $statusCode      the HTTP status code of the response
+     * @param array<string, string|string[]> $requestHeaders  the HTTP request headers
+     * @param array<string, string|string[]> $responseHeaders the HTTP response headers
+     * @param array<string, string|true>     $cacheControl    an array of parsed Cache-Control directives
+     *
+     * @return bool true if the response is cacheable, false otherwise
```

Comment:
Descriptions for `@param` and `@return` are useless here. the parameter names are already clear enough and the method description already describes the return value (and even that part could be considered useless because the method name is clear about the meaning of the returned boolean)

Reply:
adressed in https://github.com/symfony/symfony/pull/59576/commits/b3154e1c1ecf2ad2b1c7b33e95549af3c79e3f4b